reconcile/utils/saasherder.py

```diff
@@ -388,13 +388,42 @@ def _process_template(self, options):
                 return None, None
 
             if "IMAGE_TAG" not in consolidated_parameters:
-                template_parameters = template.get('parameters')
-                if template_parameters is not None:
-                    for template_parameter in template_parameters:
-                        if template_parameter['name'] == 'IMAGE_TAG':
-                            # add IMAGE_TAG only if it is required
-                            image_tag = commit_sha[:hash_length]
-                            consolidated_parameters['IMAGE_TAG'] = image_tag
+                for template_parameter in template.get("parameters", {}):
+                    if template_parameter["name"] == "IMAGE_TAG":
+                        # add IMAGE_TAG only if it is required
+                        image_tag = commit_sha[:hash_length]
+                        consolidated_parameters["IMAGE_TAG"] = image_tag
+
+            # Do this in a separate loop, since it relies on IMAGE_TAG already
+            # being calculated.
+            # This should almost never already be set in
+            # consolidated_parameters, but just in case...
+            if "REPO_DIGEST" not in consolidated_parameters:
+                for template_parameter in template.get("parameters", {}):
+                    if template_parameter["name"] == "REPO_DIGEST":
+                        try:
+                            logging.info("Generating REPO_DIGEST.")
+                            import docker
+                            docker_client = docker.from_env()
+                            reg_img = consolidated_parameters["REGISTRY_IMG"]
+                            img_tag = consolidated_parameters["IMAGE_TAG"]
+                            reg_data = docker_client.images.get_registry_data(
+                                f"{reg_img}:{img_tag}")
+                            digest = reg_data.attrs["Descriptor"]["digest"]
+                            # `digest` is of the form "sha256:abcd1234..."
+                            consolidated_parameters[
+                                "REPO_DIGEST"] = f"{reg_img}@{digest}"
+                        except Exception as e:
```

Comment:
Let's be specific about the exceptions we are handling here.

Reply:
I can peel out `KeyError`, but I'd like to keep the catch-all in any case, because I don't know what kinds of exceptions the docker lib can raise. Acceptable? (There are other spots in this file that use `Exception`...)

Reply:
We are cleaning up the pylint errors, generic exceptions being one of them. Having this here now only increases the upcoming amount of work.

Reply:
Understood. If I miss one, it'll blow up ugly instead of pretty -- I assume that's an acceptable risk?

Reply:
This is done.